rotkehlchen/accounting/structures/base.py

```diff
@@ -119,6 +127,22 @@ def __init__(
         self.notes = notes
         self.identifier = identifier
 
+        # Check that the received event type and subtype is a valid combination
+        if (
+            __debug__ and
+            (
+                (std_mapping := EVENT_CATEGORY_MAPPINGS.get(self.event_type)) is None or
+                std_mapping.get(self.event_subtype) is None
+            )
+        ) and (
+            (fallback_mapping := DEFAULT_EVENT_CATEGORY_MAPPINGS.get(self.event_type)) is None or
+            fallback_mapping.get(self.event_subtype) is None
+        ):
+            raise InvalidEventMappingError(
+                'Unexpected event type and subtype pair provided: '
+                f'{self.event_type!s}, {self.event_subtype!s}',
+            )
```

Comment:
I believe we don't need to create a custom error for this, just raise an assertion since is not an error that will be handled on production

Reply:
Resolving old conversations, thanks for taking care of it!